File: admin-ui/lib/api/types/ChatCompletionModelDto.ts
export type ChatCompletionModelDto = {
    handle: string;
    features: ChatCompletionModelDto.Features;
    reasoningType: ChatCompletionModelDto.ReasoningType;
}

export namespace ChatCompletionModelDto {
    export type Features = {
        reasoning: boolean;
        webSearch: boolean;
        reasoningEffort: boolean;
        reasoningMaxTokens: boolean;
    }

    export type ReasoningType = "NONE" | "OPTIONAL" | "REQUIRED";
}

File: src/main/kotlin/space/davids_digital/kiri/rest/dto/ChatCompletionModelDto.kt
package space.davids_digital.kiri.rest.dto

data class ChatCompletionModelDto (
    val handle: String,
    val features: Features,
    val reasoningType: ReasoningType = ReasoningType.NONE
) {
    data class Features(
        val webSearch: Boolean,
        val reasoningEffort: Boolean,
        val reasoningMaxTokens: Boolean
    )

    enum class ReasoningType {
        NONE, OPTIONAL, REQUIRED
    }
}
